lib/manager/npm/post-update/lerna.ts

```diff
@@ -34,37 +73,14 @@ export async function generateLockFiles(
     } catch (err) {
       logger.warn('Could not detect lerna version in package.json');
     }
-    lernaVersion = lernaVersion || 'latest';
-    logger.debug('Using lerna version ' + lernaVersion);
-    let params: string;
-    if (lernaClient === 'npm') {
-      if (skipInstalls === false) {
-        params = '--ignore-scripts  --no-audit';
-      } else {
-        params = '--package-lock-only --no-audit';
-      }
-    } else {
-      params = '--ignore-scripts --ignore-engines --ignore-platform';
+    if (!lernaVersion || !semver.validRange(lernaVersion)) {
+      lernaVersion = 'latest';
     }
-
-    // // istanbul ignore if
-    // if (config.dockerMapDotfiles) {
-    //   const homeDir =
-    //     process.env.HOME || process.env.HOMEPATH || process.env.USERPROFILE;
-    //   const homeNpmrc = join(homeDir, '.npmrc');
-    //   volumes.push([homeNpmrc, `/home/ubuntu/.npmrc`]);
-    // }
-    cmd.push(`${lernaClient} install ${params}`);
-    cmd.push(`npx lerna@${quote(lernaVersion)} bootstrap --no-ci -- ${params}`);
-    await exec(cmd, {
-      cwd,
-      env,
-      // docker: {
-      //   image: `renovate/${lernaClient}`,
-      //   volumes,
-      //   envVars,
-      // },
-    });
+    logger.debug('Using lerna version ' + lernaVersion);
+    cmd.push(
+      `npx lerna@${quote(lernaVersion)} bootstrap --no-ci -- ${cmdOptions}`
```

Comment:
Why not installing lerna globally too?

Reply:
It would be potentially breaking because today if you run Renovate with `binarySource=global` without `lerna` installed then it will still work thanks to `npx`. If we made it installed globally for docker and `lerna` instead of `npx lerna` then it's a different result for those users.

Reply:
good catch 🤔 Does npx reuse global installed tools? so we can install it globally on docker and npx will reuse cached version?

Reply:
Nope, it does not seem to reuse global:

```
❯ npx lerna@3.21.0 --version
npx: installed 753 in 13.004s
3.21.0

~ 15s
❯ npx lerna@3.21.0 --version
npx: installed 753 in 12.378s
3.21.0

~ 14s
❯ npm i -g lerna@3.21.0
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated mkdirp-promise@5.0.1: This package is broken and no longer maintained. 'mkdirp' itself supports promises now, please switch to that.
/usr/local/bin/lerna -> /usr/local/lib/node_modules/lerna/cli.js
+ lerna@3.21.0
added 151 packages from 99 contributors, removed 45 packages, updated 202 packages and moved 3 packages in 13.027s

~ 14s
❯ npx lerna@3.21.0 --version
npx: installed 753 in 13.237s
3.21.0

~ 15s
❯
```

Reply:
We should maybe release a new major and deprecate this `npx lerna`, but not now.